docs/sources/mimir/configure/about-versioning/index.md

```diff
@@ -0,0 +1,913 @@
+---
+aliases:
+  - ../configuring/about-versioning/
+description: Learn about guarantees for this Grafana Mimir major release.
+menuTitle: Versioning
+title: About Grafana Mimir versioning
+weight: 50
+---
+
+# About Grafana Mimir versioning
+
+This topic describes our guarantees for this Grafana Mimir major release.
+
+## Flags, configuration, and minor version upgrades
+
+Upgrading Grafana Mimir from one minor version to the next minor version should work, but we don't want to bump the major version every time we remove a configuration parameter.
+We will keep [deprecated features](#deprecated-features) in place for two minor releases.
+You can use the `deprecated_flags_inuse_total` metric to generate an alert that helps you determine if you're using a deprecated flag.
+
+These guarantees don't apply to [experimental features](#experimental-features).
+
+## Reading old data
+
+The Grafana Mimir maintainers commit to ensuring that future versions can read data written by versions within the last two years.
+In practice, we expect to be able to read data written more than two years ago, but a minimum of two years is our guarantee.
+
+## API Compatibility
+
+Grafana Mimir strives to be 100% compatible with the Prometheus HTTP API which is by default served by endpoints with the /prometheus HTTP path prefix `/prometheus/*`.
+
+We consider any deviation from this 100% API compatibility to be a bug, except for the following scenarios:
+
+- Additional API endpoints for creating, removing, modifying alerts, and recording rules.
+- Additional APIs that push metrics (under `/prometheus/api/push`).
+- Additional API endpoints for management of Grafana Mimir, such as the ring. These APIs are not included in any compatibility guarantees.
+- [Delete series API](https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series).
+
+## Experimental features
+
+Grafana Mimir is an actively developed project and we encourage the introduction of new features and capabilities.
+Not everything in each release of Grafana Mimir is considered production-ready.
+We mark as "Experimental" all features and flags that we don't consider production-ready.
+
+We do not guarantee backwards compatibility for experimental features and flags.
+Experimental configuration and flags are subject to change.
+
+The following features are currently experimental:
```

Comment:
I don't love how this is formatted - would it be possible to emit a list of experimental flags, rather than the full config reference like this?

It would also be nice to be able to group experimental features together, something like this:

> - Set Retry-After header in recoverable error responses
>   - `-distributor.retry-after-header.enabled`
>   - `-distributor.retry-after-header.base-seconds`
>   - `-distributor.retry-after-header.max-backoff-exponent`

I imagine this will require adding some extra metadata to the flags alongside the `category:"experimental"` tag.

Reply:
Sounds like a good idea, I agree. I think we could group them like so:
```
Distributor
- distributor.retry-after-header.enabled
- distributor.retry-after-header.base-seconds
- distributor.retry-after-header.max-backoff-exponent
...
```

Perhaps with some descriptions, too.
But I'm not sure what you're imagining with regards to the metadata, could you elaborate what you had in mind?

Reply:
> It would also be nice to be able to group experimental features together, something like this:

What if we simplify the grouping and extract the group name from the 1st segment of the CLI flag? For example, `-distributor.retry-after-header.enabled` group is `distributor` because extract from `-(distributor).retry-after-header.enabled`.